minio/minio
# Use a non-strict invalid bucket name check in Get and Delete object APIs

## Description
Community user reported issues with `Getobject` API with invalid bucket name in `fs` mode. Looks like we check for `bucket name` being `valid` in the case of `Get` and `Delete` object. Since we support existing buckets with invalid name, this should work.

## Motivation and Context
Community reported issue

```
mc: <DEBUG> GET /user_1/?delimiter=%2F&max-keys=1000&prefix=hosts HTTP/1.1
Host: localhost:9000
User-Agent: MinIO (linux; amd64) minio-go/v6.0.32 mc/2019-08-06T21:54:44Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20190807/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20190807T184825Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 576
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Date: Wed, 07 Aug 2019 18:48:25 GMT
Server: MinIO/DEVELOPMENT.2019-08-07T17-07-38Z
Vary: Origin
X-Amz-Request-Id: 15B8B87A323F4799
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  663.436µs

mc: <DEBUG> GET /user_1/hosts HTTP/1.1
Host: localhost:9000
User-Agent: MinIO (linux; amd64) minio-go/v6.0.32 mc/2019-08-06T21:54:44Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20190807/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20190807T184825Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 400 Bad Request
Content-Length: 310
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Date: Wed, 07 Aug 2019 18:48:25 GMT
Server: MinIO/DEVELOPMENT.2019-08-07T17-07-38Z
Vary: Origin
X-Amz-Request-Id: 15B8B87A324EABBD
X-Xss-Protection: 1; mode=block

<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>InvalidBucketName</Code><Message>The specified bucket is not valid.</Message><Key>hosts</Key><BucketName>user_1</BucketName><Resource>/user_1/hosts</Resource><RequestId>15B8B87A324EABBD</RequestId><HostId>bc0e4d1e-bacc-42eb-91ad-2d7f3eacfa8d</HostId></Error>mc: <DEBUG> Response Time:  791.532µs
```

## How to test this PR?
Create a bucket (directory) on the backend in `fs` mode.
Put object to that bucket
do `mc cat` on that object and see it fail.

## Types of changes

## Checklist:
- [ ] Fixes a regression (Don't know)
